Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -1710,6 +1858,25 @@ namespace AzToolsFramework
                 PrefabDom entityDomAfter(&domToAddDuplicatedEntitiesUnder.GetAllocator());
                 entityDomAfter.Parse(newEntityDomString.toUtf8().constData());
 
+                EntityAlias parentEntityAlias;
+                if (auto componentsIter = entityDomAfter.FindMember(PrefabDomUtils::ComponentsName);
+                    componentsIter != entityDomAfter.MemberEnd())
+                {
+                    for (auto componentIter = componentsIter->value.MemberBegin(); componentIter != componentIter->value.MemberEnd() && parentEntityAlias.empty();
+                         ++componentIter)
+                    {
+                        if (auto parentEntityIter = componentIter->value.FindMember("Parent Entity");
```

Comment:
Instead of iterating over all the components and stopping at the first occurrence of 'Parent Entity' may not be the best solution because if there's any custom component named 'Parent Entity', that will get picked and it may not be the actual alias of the parent. Instead, we can try to look for the 'TransformComponent' from the list of components and look up 'Parent Entity' within that. This will yield same behavior for all entities.

Alternatively, instead of fetching the alias of the parent from the dom, maybe you can fetch it from the actual entity itself. Similar to how we maintain a map of oldAlias to NewAlias, you can maintain a map of NewAlias to ParentAlias.

Reply:
I had a type ID check I added later for the sort order component I forgot to add here, it of course would be warranted to expand this with helper methods for checking a component's type / retrieving component doms of type / etc if we want to stick with this approach but this should be OK (verified duplication still works with the type check).